site/src/pages/WorkspacesPage/WorkspacesButton.tsx

```diff
@@ -0,0 +1,234 @@
+          overflowY: "hidden",
+        }}
+      >
+        <Avatar
+          src={template.icon}
+          fitImage
+          alt={template.display_name || "Coder template"}
+          sx={{
+            width: `${ICON_SIZE}px`,
+            height: `${ICON_SIZE}px`,
+            fontSize: `${ICON_SIZE * 0.5}px`,
+            fontWeight: 700,
+          }}
+        >
+          {template.display_name || "-"}
+        </Avatar>
+
+        <Box
+          sx={{
+            lineHeight: 1,
+            width: "100%",
+            overflow: "hidden",
+            color: "white",
+          }}
+        >
+          <Box
+            component="p"
+            sx={{
+              marginY: 0,
+              paddingBottom: 0.5,
+              overflow: "hidden",
+              textOverflow: "ellipsis",
+              whiteSpace: "nowrap",
+            }}
+          >
+            {template.display_name || "[Unnamed]"}
+          </Box>
+
+          <Box
+            component="p"
+            sx={{
+              marginY: 0,
+              fontSize: 14,
+              color: (theme) => theme.palette.text.secondary,
+            }}
+          >
+            {/*
+             * There are some templates that have -1 as their user count –
+             * basically functioning like a null value in JS. Can safely just
+             * treat them as if they were 0.
+             */}
+            {template.active_user_count <= 0
+              ? "No"
+              : template.active_user_count}{" "}
+            developer
+            {template.active_user_count === 1 ? "" : "s"}
+          </Box>
+        </Box>
+      </Box>
+    </Link>
+  );
+}
+
+export function WorkspacesButton() {
+  const organizationId = useOrganizationId();
+  const permissions = usePermissions();
+  const templatesQuery = useQuery(templates(organizationId));
+
+  // Dataset should always be small enough that client-side filtering should be
+  // good enough. Can swap out down the line if it becomes an issue
+  const [searchTerm, setSearchTerm] = useState("");
+  const processed = sortTemplatesByUsersDesc(
+    templatesQuery.data ?? [],
+    searchTerm,
+  );
+
+  let emptyState: ReactNode = undefined;
+  if (templatesQuery.data?.length === 0) {
+    emptyState = (
+      <EmptyState
+        message="No templates yet"
+        cta={
+          <Link to="/templates" component={RouterLink}>
+            Create one now.
+          </Link>
+        }
+      />
+    );
+  } else if (processed.length === 0) {
+    emptyState = <EmptyState message="No templates match your text" />;
+  }
+
+  return (
+    <PopoverContainer
+      // Stopgap value until bug where string-based horizontal origin isn't
+      // being applied consistently can get figured out
+      originX={-115}
+      originY="bottom"
+      sx={{ display: "flex", flexFlow: "column nowrap" }}
+      anchorButton={
+        <Button startIcon={<AddIcon />} variant="contained">
+          Create Workspace&hellip;
```

Comment:
Why the ellipsis?

Reply:
Just an UI concern – it just helps communicate that there's no immediate action when you click the button, and you'll see more options when you do.

Kayla had a thread about this recently:
https://codercom.slack.com/archives/CJURPL8DN/p1694725326331259